src/azure-cli-core/azure/cli/core/mock.py

```diff
@@ -32,17 +32,23 @@ def __init__(self, commands_loader_cls=None, random_config_dir=False, **kwargs):
             self.env_patch = patch.dict(os.environ, {'AZURE_CONFIG_DIR': config_dir})
             self.env_patch.start()
 
+            # Always copy command index to avoid initializing it again
+            files_to_copy = ['commandIndex.json']
             # In recording mode, copy login credentials from global config dir to the dummy config dir
             if os.getenv(ENV_VAR_TEST_LIVE, '').lower() == 'true':
-                if os.path.exists(GLOBAL_CONFIG_DIR):
```

Comment:
Even if `GLOBAL_CONFIG_DIR` doesn't exist, no error will occur as `FileNotFoundError` is ignored.